helpers/collection/collection.go

```diff
@@ -8,83 +8,101 @@ import (
 	"github.com/go-pg/pg/orm"
 )
 
-const defaultOrdersLimit = 25
+// DefaultItemsLimit per page limit
+var DefaultItemsLimit = 50
+
+// SeparatorOrder
+var SeparatorOrder = ","
+
+type (
+	// Former interface
+	Former interface {
+		ApplyFilter(*Options) error
+		ApplyPager(*Options) error
+		ApplySorter(*Options) error
+	}
 
-// Filter interface
-type Filter interface {
-	Pager() Paginate
-	Apply(*orm.Query) error
-}
+	// Fetcher closure to fetch and format Items
+	Fetcher = func(*Options) (Items, error)
 
-// Paginate is part of filter structure
-type Paginate struct {
-	Offset int `query:"offset" form:"offset" json:"offset"`
-	Limit  int `query:"limit" form:"limit" json:"limit"`
-}
+	// Items is formatted response of models
+	Items = []interface{}
 
-// Pager returns Paginate-struct
-func (p Paginate) Pager() Paginate {
-	var limit = defaultOrdersLimit
+	// Collection response answer
+	Collection struct {
+		Total  int   `json:"total"`
+		Offset int   `json:"offset"`
+		Items  Items `json:"items"`
+	}
 
-	if p.Limit > 0 {
-		limit = p.Limit
+	// Options to Format list-response
+	Options struct {
+		Query   *orm.Query
+		Fetcher Fetcher
+		Former  Former
 	}
+)
 
-	return Paginate{
-		Offset: p.Offset,
-		Limit:  limit,
+// ResponseQuery collection as response-answer (Collection)
+func ResponseQuery(ctx web.Context, opts Options) error {
+	if opts.Query == nil || opts.Former == nil || opts.Fetcher == nil {
+		return nil
 	}
-}
 
-// Collection response answer
-type Collection struct {
-	Total  int   `json:"total"`
-	Offset int   `json:"offset"`
-	Items  Items `json:"items"`
-}
+	var (
+		err      error
+		response Collection
+	)
+
+	if err = opts.Former.ApplyFilter(&opts); err != nil {
+		ctx.Logger().Error(err.Error())
+		return errors.NewError(http.StatusBadRequest, err.Error())
+	}
 
-// ModelsFetcher closure to fetch and format Items
-type ModelsFetcher = func(*orm.Query) (Items, error)
+	if response.Total, err = opts.Query.Count(); err != nil {
+		ctx.Logger().Error(err.Error())
+		return errors.NewError(http.StatusInternalServerError, err.Error())
+	}
+
+	if err = opts.Former.ApplyPager(&opts); err != nil {
+		ctx.Logger().Error(err.Error())
+		return errors.NewError(http.StatusBadRequest, err.Error())
+	}
 
-// Items is formatted response of models
-type Items = []interface{}
+	if err = opts.Former.ApplySorter(&opts); err != nil {
+		ctx.Logger().Error(err.Error())
+		return errors.NewError(http.StatusBadRequest, err.Error())
+	}
 
-// Options to Format list-response
-type Options struct {
-	Query   *orm.Query
-	Fetcher ModelsFetcher
-	Filter  Filter
+	return format(ctx, &opts, response)
 }
 
-// Format collection as response-answer (Collection)
-func Format(ctx web.Context, opts Options) error {
-	if opts.Query == nil || opts.Filter == nil || opts.Fetcher == nil {
+// ResponseSimple collection as response-answer (Collection)
+func ResponseSimple(ctx web.Context, opts Options) error {
```

Comment:
Like I said, we don't have cases for this method.
Thats why, I do not see any sense in this method.

Reply:
my own idea